tests/samples_tests/smoke_tests/test_benchmark_app.py

```diff
@@ -161,3 +162,64 @@ def test_out_of_tensor_size_range_npy_multibatch(sample_language, device, cache,
     # benchmark_app reads batch from model or cmd, not from npy.
     # benchmark_app still verifyes npy shape for python impl.
     verify(sample_language, device, inp=inp, cache=cache, tmp_path=tmp_path, batch='2')
+
+@pytest.mark.parametrize('sample_language', ['C++', 'Python'])
+@pytest.mark.parametrize('device', get_devices())
+@pytest.mark.parametrize('pv_in_tensor_names', [{'87', 'input.1'}, {'87', 'input.1', 'pixel_values'}, {''}])
+@pytest.mark.parametrize('pv_out_tensor_names', [{''}])
+@pytest.mark.parametrize('am_in_tensor_names', [{'877', 'attention_mask'}])
+@pytest.mark.parametrize('am_out_tensor_names', [{''}])
+def test_input_tensor_with_multiple_names(sample_language, device, pv_in_tensor_names, pv_out_tensor_names, am_in_tensor_names, am_out_tensor_names, cache, tmp_path):
+    '''
+    Tests the ability of finding a proper tensor name by an argument parsed from cmd -shape or -data_shape
+    '''
+    pv_param_shape = [-1, -1, -1, -1]
+    pv_data_shape = [1, 3, 224, 224]
+    pv_param_node = opset.parameter(pv_param_shape, ov.Type.f32, name="pixel_values")
+    pv_param_node.output(0).set_names(pv_in_tensor_names)
+    pv_cvt_node = opset.convert(pv_param_node, ov.Type.f16)
+    pv_result_node = opset.result(pv_cvt_node, name='pv_result')
+    pv_result_node.output(0).set_names(pv_out_tensor_names)
+
+    am_param_shape = [-1, -1]
+    am_data_shape = [1, 77]
+    am_param_node = opset.parameter(am_param_shape, ov.Type.i64, name="attention_mask")
+    am_param_node.output(0).set_names(am_in_tensor_names)
+    am_cvt_node = opset.convert(am_param_node, ov.Type.i32)
+    am_result_node = opset.result(am_cvt_node, name='am_result')
+    am_result_node.output(0).set_names(am_out_tensor_names)
+
+    model = ov.Model([pv_result_node, am_result_node], [pv_param_node, am_param_node], 'model_with_multi_name_input_tensor')
+    data_shape = f'pixel_values{pv_data_shape},attention_mask{am_data_shape}'
+    # We intentionally set shape here instead of data_shape to catch an assert from benchmark_app,
+    # if one of command line input shapes has not been captured correctly.
+    verify(sample_language, device, shape=data_shape, model=model, inp=None, cache=cache, tmp_path=tmp_path, batch=None)
+
+@pytest.mark.parametrize('sample_language', ['C++', 'Python'])
+@pytest.mark.parametrize('device', get_devices())
+@pytest.mark.parametrize('in_node_name, in_tensor_names, out_node_name, out_tensor_names', [
+    ('input', {'actual_input'}, 'actual_input', {'input'}),
+])
+def test_input_output_tensor_name_collision(sample_language, device, in_node_name, in_tensor_names, out_node_name, out_tensor_names, cache, tmp_path):
+    '''
+    Tests the order of finding a proper tensor name by an argument parsed from cmd -shape or -data_shape
```

Comment:
The test actually tries `-iop` instead of `-shape` or `-data_shape`. They follow similar code paths so they are interchangeable in terms of test coverage and it's actually a good thing to test diverse parameters.
Please update this and `test_input_tensor_with_multiple_names()` so the docstrings describe what's actually tested. There are two ways to achieve the alignment:
1. Add new tests (or `@pytest.mark.parametrize` if possible) for `-shape` and `-data_shape`
2. Just update docstrings limiting to the actual arg used and add another sentence explaining that other flags implementation is tested implicitly.

I'd prefer the first option, but it's up to you.

Reply:
I have chosen the 2nd option just because of this:
https://github.com/openvinotoolkit/openvino/blob/c6a691205701b8fb801a98daec99ce42ddcf2845/samples/cpp/benchmark_app/main.cpp#L751 
`outputs_info` is passed to this function only from this place, under `if (-iop)` condition.
Considering this, I think there are no corner casesthat  might be tested if we have only `inputs_info`